packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -75,6 +75,7 @@ export const GraphNG: React.FC<GraphNGProps> = ({
     // X is the first field in the alligned frame
     const xField = alignedFrame.fields[0];
     if (xField.type === FieldType.time) {
+      builder.xScaleIsTimeRange = true;
       builder.addScale({
         scaleKey: 'x',
         isTime: true,
```

Comment:
Maybe instead of setting this flag explicitly we could set it when adding scale that `isTime`? Not sure we will fallback to data set time range in time scenario. WDYT?

Reply:
yea, having both seem strange